src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -472,13 +492,46 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             let preferredConnection: KernelConnectionMetadata | undefined;
             // Don't attempt preferred kernel search for interactive window, but do make sure we
             // load all our controllers for interactive window
+            const notebookMetadata = getNotebookMetadata(document);
             if (document.notebookType === JupyterNotebookView) {
-                preferredConnection = await this.kernelFinder.findKernel(
-                    document.uri,
-                    getNotebookMetadata(document),
-                    preferredSearchToken.token
+                const resourceType = getResourceType(document.uri);
+                const isPythonNbOrInteractiveWindow =
+                    isPythonNotebook(notebookMetadata) || resourceType === 'interactive';
+                const preferredInterpreter =
+                    isPythonNbOrInteractiveWindow && this.extensionChecker.isPythonExtensionInstalled
+                        ? await this.interpreters.getActiveInterpreter(document.uri)
+                        : undefined;
+
+                ({ preferredConnection } = await this.findPreferredKernelExactMatch(
```

Comment:
This is returning { preferredConnection } since I changed the function to also support returning the full ranked list if we want that later.

Reply:
In the future we might do something like returning top X matches as suggestions if we don't have an exact match. But per convo with Jim leaving that out now. Doesn't make as much sense if we don't have an API distinction between suggestions and exact matches.